File: scribbleshare-commons/src/main/java/net/stzups/scribbleshare/data/objects/canvas/canvasObject/LineCanvasObject.java
package net.stzups.scribbleshare.data.objects.canvas.canvasObject;

import io.netty.buffer.ByteBuf;

public class LineCanvasObject extends CanvasObject {
    private static class Point {
        private final short x;
        private final short y;

        private Point(ByteBuf byteBuf) {
            this.x = byteBuf.readShort();
            this.y = byteBuf.readShort();
        }

        private void serialize(ByteBuf byteBuf) {
            byteBuf.writeShort(x);
            byteBuf.writeShort(y);
        }
    }

    private final Point[] points;

    public LineCanvasObject(ByteBuf byteBuf) {
        super(byteBuf);
        points = new Point[byteBuf.readUnsignedByte()];
        for (int i = 0; i < points.length; i++)  {
            points[i] = new Point(byteBuf);
        }
    }

    @Override
    public void serialize(ByteBuf byteBuf) {
        byteBuf.writeByte((byte) points.length);
        for (Point point : points) {
            point.serialize(byteBuf);
        }
    }
}

File: scribbleshare-app/scribbleshare-frontend/src/scripts/canvas/canvasObject/LineCanvasObject.js
import CanvasObject from "./CanvasObject.js";

export default class LineCanvasObject extends CanvasObject {
    constructor(reader) {
        super(reader);
        this.points = [];
        let length = reader.readUint8();
        for (let i = 0; i < length; i++) {
             this.points[i] = new Point(reader);
        }
    }

    serialize(writer) {
        writer.writeUint8();
        this.points.forEach((point) => {
            point.serialize(writer);
        })
    }
}

class Point {
    constructor(reader) {
        this.x = reader.readInt16();
        this.y = reader.readInt16();
    }

    serialize(writer) {
        writer.writeInt16(this.x);
        writer.writeInt16(this.y);
    }
}